### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by sophialee0416**
### Platforms Impacted
- [ ] iOS
- [ ] macOS

### Description of changes

Updating readme to reflect version updates in Xcode and Swift

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1272)

### CODE REVIEW

**Review by mischreiber — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ Fluent UI Apple contains native UIKit and AppKit controls aligned with [Microsof
 #### Requirements
 
 - iOS 14+ or macOS 10.15+
```
Comment (mischreiber) [inline README.md:17]: Don't we require iOS 15 now too?

Comment (mischreiber) [reply README.md:17]: Or did we not merge that yet?

Comment (huwilkes) [inline README.md:17]: The build settings for the Mac app says 12.3, but the xcconfig file says 10.15. Do we need to align those? Doesn't have to happen in this PR, either way.

Comment (huwilkes) [reply README.md:17]: We haven't dropped support for 14 quite yet, I don't think.

**Review by huwilkes — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (sophialee0416) [reply README.md:17]: That's a good question! I'm not quite sure... I can make those changes if need be. @mischreiber?

**Comment by sophialee0416** [comment]
Merging this change for now, but we can always update the readme for iOS 14 + macOS 12.3 at a later PR.

**Pull request merged**